plugins/nf-prov/src/main/nextflow/prov/BcoRenderer.groovy

```diff
@@ -0,0 +1,238 @@
+                if( taskLookup[path] )
+                    return
+
+                workflowInputs << path
+            }
+        }
+
+        return workflowInputs
+    }
+
+    /**
+     * Normalize local paths to remove environment-specific directories.
+     *
+     * @param path
+     */
+    private String normalizePath(Path path) {
+        // TODO: append raw github URL to git assets
+        path.toUriString()
+            .replace(metadata.workDir.toUriString(), '/work')
+            .replace(metadata.projectDir.toUriString(), '')
+            .replace(metadata.launchDir.toUriString(), '')
+    }
+
+    @Override
+    void render(Session session, Set<TaskRun> tasks, Map<Path,Path> workflowOutputs, Path path) {
+        if( !path.mkdirs() )
+            throw new AbortOperationException("Unable to create directory for provenance manifest: ${path.toUriString()}")
+
+        // get workflow inputs
+        final taskLookup = getTaskLookup(tasks)
+        final workflowInputs = getWorkflowInputs(tasks, taskLookup)
+
+        // get workflow metadata
+        this.metadata = session.workflowMetadata
+        final manifest = metadata.manifest
+        final manifestExtra = session.config.navigate('prov.metadata', [:]) as Map
+
+        final formatter = DateTimeFormatter.ISO_INSTANT
+        final dateCreated = formatter.format(metadata.start)
+        final authors = (manifest.author ?: '').tokenize(',')
+
+        // render BCO manifest
+        final bcoPath = path.resolve('bco.json')
+        final bco = [
+            "object_id": null,
+            "spec_version": null,
+            "etag": null,
+            "provenance_domain": [
+                "name": manifest.name,
+                "version": manifest.version,
+                "created": dateCreated,
+                "modified": dateCreated,
+                "contributors": authors.collect( name -> [
+                    "contribution": ["authoredBy"],
+                    "name": name
+                ] ),
+                "license": manifestExtra.license
+            ],
+            "description_domain": [
+                "pipeline_steps": tasks.collect { task -> [
+                    "step_number": task.id,
+                    "name": task.hash.toString(),
+                    "description": task.name,
+                    "input_list": task.getInputFilesMap().collect { name, source -> 
+                        normalizePath(source)
+                    },
+                    "output_list": getTaskOutputs(task).collect { source ->
+                        normalizePath(source)
+                    }
+                ] },
+            ],
+            "execution_domain": [
+                "script": [ normalizePath(metadata.scriptFile) ],
+                "script_driver": "nextflow"
+            ],
+            "io_domain": [
+                "input_subdomain": workflowInputs.collect { source ->
+                    [ "uri": normalizePath(source) ]
```

Comment:
Checking the docs [here](https://docs.biocomputeobject.org/io-domain/#271-input-subdomain-input_subdomain), it seems there is supposed to be one hierarchy level more per input here, with something similar to:
```json
{
    "uri": {
        "filename": "Hepatitis C virus type 1b complete genome", 
        "uri": "http://www.ncbi.nlm.nih.gov/nuccore/5420376",
        "access_time": "2017-01-24T09:40:17-0500"
    }
},
``` 
... per input.